core/src/test/java/org/mule/processor/chain/BlockingProcessorExecutorTestCase.java

```diff
@@ -0,0 +1,137 @@
+package org.mule.processor.chain;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.*;
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+import org.mule.DefaultMuleMessage;
+import org.mule.MessageExchangePattern;
+import org.mule.api.MessagingException;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.MuleMessage;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.processor.ProcessorExecutor;
+import org.mule.execution.MessageProcessorExecutionTemplate;
+import org.mule.processor.BlockingProcessorExecutor;
+import org.mule.tck.SensingNullMessageProcessor;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.size.SmallTest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+@SmallTest
+@RunWith(MockitoJUnitRunner.class)
+public class BlockingProcessorExecutorTestCase extends AbstractMuleTestCase
+{
+
+    @Mock(answer = RETURNS_DEEP_STUBS)
+    protected MuleContext muleContext;
+
+    @Mock
+    protected MuleEvent event;
+
+    @Mock
+    protected MessageProcessorExecutionTemplate executionTemplate;
+
+    protected static String A = "a";
+    protected static String B = "b";
+    protected static String C = "c";
+    protected static String RESULT = A + B + C;
+
+    protected SensingNullMessageProcessor mp1 = new SensingNullMessageProcessor(A);
```

Comment:
I can change this to A_STRING or something, but it doesn't add any value and makes things more verbose.